src/commands/admin/announce.ts

```diff
@@ -1,42 +1,44 @@
-import { botMasterPerms } from '@lib/permissions';
-import { channelParser } from '@lib/arguments';
-import { TextChannel, Message } from 'discord.js';
+import { BOTMASTER_PERMS } from '@lib/permissions';
+import { TextChannel, Message, ApplicationCommandPermissionData, CommandInteraction, ApplicationCommandOptionData } from 'discord.js';
 import { CHANNELS } from '@root/config';
 import { Command } from '@lib/types/Command';
 
 export default class extends Command {
 
 	description = 'Sends an announcement from Sage to a specified channel or announcements if no channel is given.';
 	usage = '[channel] | <content>';
-
-	async permissions(msg: Message): Promise<boolean> {
-		return await botMasterPerms(msg);
-	}
-
-	async run(msg: Message, [channel, content]: [TextChannel, string]): Promise<Message> {
+	tempPermissions: ApplicationCommandPermissionData[] = [BOTMASTER_PERMS];
+
+	options: ApplicationCommandOptionData[] = [{
+		name: 'content',
+		description: 'announcement content',
+		type: 'STRING',
+		required: true
+	},
+	{
+		name: 'image',
+		description: 'announcement image url',
+		type: 'STRING',
+		required: false
+	}]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const channel = interaction.guild.channels.cache.get(CHANNELS.ANNOUNCEMENTS) as TextChannel;
```

Comment:
Probably a low priority thing (if we agree on this we can open it as an issue for later) but we should be able to specify the channel to send the announcement in. Could be handy once we start adding role dropdowns to the main server.

Reply:
good idea, we should talk abt that

Reply:
Something to discuss on Friday post-presentation I reckon :)

Reply:
Before slashification, it had the option to do this. You can have an argument of type `CHANNEL`, so it shouldn't be too hard.